spot_wrapper/spot_arm.py

```diff
@@ -0,0 +1,609 @@
+            if not success:
+                self._logger.info(msg)
+                return False, msg
+            else:
+                # Duration in seconds.
+                traj_duration = data.duration
+
+                # first point on trajectory
+                wrench0 = self.create_wrench_from_forces_and_torques(
+                    data.forces_pt0, data.torques_pt0
+                )
+                t0 = seconds_to_duration(0)
+                traj_point0 = trajectory_pb2.WrenchTrajectoryPoint(
+                    wrench=wrench0, time_since_reference=t0
+                )
+
+                # Second point on the trajectory
+                wrench1 = self.create_wrench_from_forces_and_torques(
+                    data.forces_pt1, data.torques_pt1
+                )
+                t1 = seconds_to_duration(traj_duration)
+                traj_point1 = trajectory_pb2.WrenchTrajectoryPoint(
+                    wrench=wrench1, time_since_reference=t1
+                )
+
+                # Build the trajectory
+                trajectory = trajectory_pb2.WrenchTrajectory(
+                    points=[traj_point0, traj_point1]
+                )
+
+                # Build the trajectory request, putting all axes into force mode
+                arm_cartesian_command = arm_command_pb2.ArmCartesianCommand.Request(
+                    root_frame_name=data.frame,
+                    wrench_trajectory_in_task=trajectory,
+                    x_axis=arm_command_pb2.ArmCartesianCommand.Request.AXIS_MODE_FORCE,
+                    y_axis=arm_command_pb2.ArmCartesianCommand.Request.AXIS_MODE_FORCE,
+                    z_axis=arm_command_pb2.ArmCartesianCommand.Request.AXIS_MODE_FORCE,
+                    rx_axis=arm_command_pb2.ArmCartesianCommand.Request.AXIS_MODE_FORCE,
+                    ry_axis=arm_command_pb2.ArmCartesianCommand.Request.AXIS_MODE_FORCE,
+                    rz_axis=arm_command_pb2.ArmCartesianCommand.Request.AXIS_MODE_FORCE,
+                )
+                arm_command = arm_command_pb2.ArmCommand.Request(
+                    arm_cartesian_command=arm_cartesian_command
+                )
+                synchronized_command = (
+                    synchronized_command_pb2.SynchronizedCommand.Request(
+                        arm_command=arm_command
+                    )
+                )
+                robot_command = robot_command_pb2.RobotCommand(
+                    synchronized_command=synchronized_command
+                )
+
+                # Send the request
+                cmd_id = self._robot_command_client.robot_command(robot_command)
+                self._logger.info("Force trajectory command sent")
+                self.wait_for_arm_command_to_complete(cmd_id)
+        except Exception as e:
+            return False, f"Exception occured during arm movement: {e}"
+
+        return True, "Moved arm successfully"
+
+    def gripper_open(self) -> typing.Tuple[bool, str]:
```

Comment:
When I run `gripper_open, gripper_close, gripper_angle_open` I get an error while the system gets stuck on `block_until_arm_arrives` forever. Is this just an error on my side? the `arm._unstow, arm_carry, arm_stow` functions don't have the same issue.

Reply:
`block_until_arm_arrives` doesn't have any checks for gripper commands

Reply:
Looks like I need to use https://dev.bostondynamics.com/protos/bosdyn/api/proto_reference.html#clawgrippercommand-feedback-status and write something similar to the blocking manipulation command.